test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -141,6 +141,38 @@
         ;; now the pool is flushed, so the constants should be cleared
         (is (true? (verify-no-constants pool-context 4)))))))
 
+(deftest ^:integration admin-api-multiple-flush-jruby-pool-test
+  (testing "Flushing the jruby pool multiple times results in all new JRuby instances"
+    (bootstrap/with-puppetserver-running
+      app
+      {:puppet-admin {:client-whitelist ["localhost"]}
+       :jruby-puppet {:max-active-instances 4}}
+      (let [jruby-service (tk-app/get-service app :JRubyPuppetService)
+            context (tk-services/service-context jruby-service)
+            pool-context (:pool-context context)
+            times-to-flush 3]
+        ;; set a ruby constant in each instance so that we can recognize them
+        (is (true? (set-constants-and-verify pool-context 4)))
+        (let [times-flushed (atom 0)
+              _ (add-watch (:pool-agent pool-context)
+                           :flush-callback
+                           (fn [k _ _ _]
+                             (when (= k :flush-callback)
+                               (swap! times-flushed inc))))
+              wait-until (+ (System/currentTimeMillis) 300000)]
```

Comment:
I put the 5 minute time limit on this to ensure that the test doesn't run in an infinite loop if the flushes never happen.  5 minutes is fairly arbitrary but I figured it should be a reasonable upper bound.